protocol/l2-blob.md

```diff
@@ -0,0 +1,199 @@
+that rely on BLOBs.
+
+The goal of this specification is to **support L2 BLOB transactions in the OP Stack**. This would allow L3 solutions,
+which settle on L2, to have an enshrined 4844-compatible DA layer that they can use directly, without needing to
+integrate third-party DA providers or deal with the security risks associated with DA bridges. Additionally, the
+applications mentioned above could migrate to L2 with minimal costs.
+
+Furthermore, this spec proposes using
+[Alt-DA](https://github.com/ethereum-optimism/specs/blob/main/specs/experimental/alt-da.md) to upload L2 BLOBs while
+still using L1 DA for L2 calldata. This approach, referred to as a “hybrid DA L2”, combines the best features of
+different DA solutions. This allows
+users and applications of an L2 to choose between L1 DA and alt-DA for different types of transaction data within the
+same network, without the need to maintain multiple L2s. Specifically, users can upload and store non-financial data
+at a very low cost using L2 BLOBs and Alt-DA, while still conducting critical financial data using L2 calldata and
+L1 DA. In some cases, these two types of data may even occur within the same transaction. Here are a few potential
+scenarios:
+
+- While the Optimism mainnet continues to use L1 DA for uploading calldata, multiple app-specific or game-focused
+L3s settled on it can directly use the enshrined 4844-compatible L2 DA layer, benefiting from easy integration, robust
+security, and lower costs.
+- Users might use a platform like Decentralized Twitter primarily for social networking (non-financial), while
+also sending payments (financial) to other users within the same application.
+
+The following diagram illustrates the transaction data flow for a hybrid DA L2:
+
+```mermaid
+flowchart
+    A[Users] -->|Non-financial Tx Using BLOB| B(L2)
+    A[Users] -->|Financial Tx Using Calldata| B(L2)
+    B -->|L2 BLOB| C(Alt-DA)
+    B -->|L2 Calldata| D(L1 DA)
+```
+
+# Proposed Solution
+
+<!-- A high level overview of the proposed solution.
+When there are multiple alternatives there should be an explanation
+of why one solution was picked over other solutions.
+As a rule of thumb, including code snippets (except for defining an external API)
+is likely too low level. -->
+
+## Enabling BLOB Transactions in L2 EL
+
+The interface and implementation of L2 EL should remain consistent with L1 EL to ensure seamless migration of
+applications. Note that while BLOBs are gossiped within the L1 P2P network, for enshrined BLOB DA support in L2, the
+BLOBs should be sent directly to the L2 sequencer.
+
+## Uploading BLOB to Alt-DA
+
+The sequencer is responsible for uploading BLOBs to the Alt-DA layer. When the
+CL (op-node) of the sequencer receives the payload from its EL via the engine API, it should inspect the envelope for any `BlobsBundle`
+and upload them to the Alt-DA. Only after ensuring successful BLOB uploads can the sequencer upload the block data
```

Comment:
> Only after ensuring successful BLOB uploads can the sequencer upload the block data to the L1 DA.

Sequencer is not responsible to upload block data to L1 DA, it's the batcher that's responsible for this.

Reply:
According to the OP [spec](https://github.com/ethereum-optimism/specs/blob/main/specs/glossary.md#sequencer), `The sequencer is a privileged actor, which receives L2 transactions from L2 users, creates L2 blocks using them, which it then submits to data availability provider (via a batcher). It also submits output roots to L1.` So IIUC, a sequence includes full node + batcher + proposer.

Reply:
OK, it's a more general concept.

Reply:
OK, it's a more general concept.